### PR TITLE
add X-Trino-Prepared-Statement to support SQL placeholder

### PR DESCRIPTION
**Pull request by likyh**
```ts
client.execute({
        query: `EXECUTE query0`,
        prepares: [ 'SELECT xxxxxx' ],
});

client.execute({
        query: `EXECUTE query0 USING 1,2,3`,
        prepares: [ 'SELECT * FROM user WHERE id=? AND ……' ],
});
```

add X-Trino-Prepared-Statement to support SQL placeholder

### CODE REVIEW

**Comment by tagomoris** [comment]
Documents: [1](https://trino.io/docs/current/sql/prepare.html), [2](https://trino.io/docs/current/sql/execute.html), [3](https://docs.starburst.io/356-e/develop/client-protocol.html)
The document's description is not clear to me, so I want to clarify its requirement: Should be specify the query and `X-Trino-Prepared-Statement` like below?:
* query: `EXECUTE query0 USING 1,2,3`
* X-Trino-Prepared-Statement: query0=SELECT * FROM users WHERE col1=? AND col2=? AND col3=?,query1=....`

I'm sure it can't be parsed correctly for non-quoted SQL queries...

File: lib/presto-client/index.js [hunk 1]
```diff
@@ -198,6 +198,10 @@ Client.prototype.statementResource = function(opts) {
     if (opts.schema || this.schema) {
         header[client.headers.SCHEMA] = opts.schema || this.schema;
     }
+    if (opts.prepares) {
+        header[client.headers.PREPARE] = opts.prepares.map((s, index) => 'query' + index + '=' + s).join(',');
```
Comment (tagomoris) [inline lib/presto-client/index.js:202]: Could you add a check if `client.headers.PREPARE` is not defined (for Presto not Trino)?

**Comment by likyh** [comment]
> Documents: [1](https://trino.io/docs/current/sql/prepare.html), [2](https://trino.io/docs/current/sql/execute.html), [3](https://docs.starburst.io/356-e/develop/client-protocol.html)
> The document's description is not clear to me, so I want to clarify its requirement: Should be specify the query and `X-Trino-Prepared-Statement` like below?:
> 
> * query: `EXECUTE query0 USING 1,2,3`
> * X-Trino-Prepared-Statement: query0=SELECT * FROM users WHERE col1=? AND col2=? AND col3=?,query1=....`
> 
> I'm sure it can't be parsed correctly for non-quoted SQL queries...

Yes. It's cannot use like these:
```js
client.execute({
        query: `PREPARE query0 FROM SELECT * FROM nation;`,
});
client.execute({
        query: `EXECUTE query0`,
});
# It cannot use in Http but can use in Cli. (session is create in each http request)
```

```js
client.execute({
        query: `PREPARE query0 FROM SELECT * FROM nation; EXECUTE query0`,
});
# not support semicolon
```

It can use like this:
query: urlencode("EXECUTE query0 USING 1,2,'abc'")
X-Trino-Prepared-Statement: query0=SELECT * FROM users WHERE col1=? AND col2=? AND col3=?`


What's your 「non-quoted SQL queries」 means?

**Comment by tagomoris** [comment]
> What's your 「non-quoted SQL queries」 means?

The [document](https://docs.starburst.io/356-e/develop/client-protocol.html) says "A comma-separated list of the name=value pairs" about `X-Trino-Prepared-Statement`, but, as you know, SQLs can contain commas. So I'm unsure what will happen if the value of `X-Trino-Prepared-Statement` is like below:
```
X-Trino-Prepared-Statement: query0=SELECT a, b, c FROM users, query1=SELECT d, e FROM items
```

Can Trino (and Presto) parse this value correctly without quoting? I guess my confusion is coming from the unclear expression of the document (What is it?):
> and the values are keys that identify the executable form of the named prepared statements
```

**Comment by tagomoris** [comment]
Hmm, Presto also supports `X-Presto-Prepared-Statement` https://prestodb.io/docs/current/develop/client-protocol.html

**Comment by likyh** [comment]
> > What's your 「non-quoted SQL queries」 means?
> 
> The [document](https://docs.starburst.io/356-e/develop/client-protocol.html) says "A comma-separated list of the name=value pairs" about `X-Trino-Prepared-Statement`, but, as you know, SQLs can contain commas. So I'm unsure what will happen if the value of `X-Trino-Prepared-Statement` is like below:
> 
> ```
> X-Trino-Prepared-Statement: query0=SELECT a, b, c FROM users, query1=SELECT d, e FROM items
> ```
> 
> Can Trino (and Presto) parse this value correctly without quoting? I guess my confusion is coming from the unclear expression of the document (What is it?):
> 
> > and the values are keys that identify the executable form of the named prepared statements

Ok, I understand. `prepares` need urlencode to avoid this mistakes.

```
X-Trino-Prepared-Statement: query0=urlencode(SELECT a, b, c FROM users),query1=urlencode(SELECT d, e FROM items)
X-Trino-Prepared-Statement: query0=SELECT%20a%2C%20b%2C%20c%20FROM%20users,query1=SELECT%20d%2C%20e%20FROM%20items
```

I'll update this PR

**Comment by likyh** [comment]
> Hmm, Presto also supports `X-Presto-Prepared-Statement` https://prestodb.io/docs/current/develop/client-protocol.html

I haven't use Presto, maybe code need some test in presto.

**Comment by tagomoris** [comment]
I missed the comments and changes after my last review comments. Sorry for that.

**Review by tagomoris — APPROVED** [review]

**Pull request merged**